src/debug/telemetry/telemetry2_thread_info.c

```diff
@@ -0,0 +1,349 @@
+/* SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) 2024 Intel Corporation
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <stdio.h>
+#include <soc.h>
+#include <adsp_debug_window.h>
+
+#include "telemetry2_slot.h"
+
+LOG_MODULE_REGISTER(thread_info);
+
+/* to adsp backend header, data to be shared with user space client*/
+#define THREAD_INFO_MAX_THREADS 16
+
+struct thread_info {
+	char name[14];
```

Comment:
where does the 14 come from?

Reply:
Just a nice number to get 16-byte thread records. The rest of the thread name is truncated. Actually there are no names available unless CONFIG_THREAD_NAME=y . That should maybe be selected by SOF_TELEMETRY2_THREAD_INFO option.

Reply:
CONFIG_THREAD_NAME is a Zephyr Kconfig option.

Reply:
Oh, its actually a Kconfig variable THREAD_MAX_NAME_LEN, and defaults to 32. However, with the static data structure I did not want to waste memory for maximum thread name length, when 14 is usually enough. I think that other name length definitions are there for similar reasons. In any case, the in the new circular buffer based approach I intend to make the name length dynamic, not static.